web/js/js.gradle

```diff
@@ -15,13 +12,12 @@ dependencies {
 }
 
 /**
- * Provides a jar full of static js/html/css that can be served from jetty
+ * Attach the static js/html/css to the main jar so that can be served from jetty
  */
-def ideClientJsJar = tasks.register('ideClientJsJar', Jar) {
-    from configurations.js
-}
+tasks.named(JavaPlugin.JAR_TASK_NAME, Jar, { jar ->
+    jar.from(configurations.js)
```

Comment:
I assume that when built this way, gradle doesn't consider those dependencies to be transitive? That was what I was trying to avoid to begin with by making this new configuration.

Reply:
We are "skipping" the dependency transitivity by attach the outputs directly to the jar afaict. Maybe @JamesXNelson has a better way of doing this? I'm trying to avoid declaring a dependency on new configuration, https://docs.gradle.org/current/userguide/cross_project_publications.html#sec:simple-sharing-artifacts-between-projects:

> Declaring a dependency on an explicit target configuration is not recommended if you plan to publish the component which has this dependency: this would likely lead to broken metadata.

(Variant-aware sharing of artifacts between projects seems complex.)

```
$ ./gradlew web-js:jar web-js:generateMetadataFileForMavenJavaPublication web-js:generatePomFileForMavenJavaPublication
$ jar tf web/js/build/libs/deephaven-web-js-0.10.0.jar
$ cat web/js/build/publications/mavenJava/pom-default.xml
$ cat web/js/build/publications/mavenJava/module.json
```

Is how I examined the output.

Reply:
This will also include the .java files.  For better or worse...

Since we're open source it doesn't _really_ matter. A couple KB of noise in the file.

If we wanted to _replace_ the java jar with the js jar, we'd have to get a little more creative with the publication configuration... possible, but I don't really see any value in doing so.  _this is fine_.

Reply:
Note: there *aren't* any .java files :)